component/src/main/java/org/wso2/extension/siddhi/execution/streamingml/clustering/kmeans/KMeansIncrementalSPExtension.java

```diff
@@ -0,0 +1,305 @@
+        description = "Performs K-Means clustering on a streaming data set. Data points can be of any dimension and " +
+                "the dimensionality is calculated from number of parameters. " +
+                "All data points to be processed by an instance of class Clusterer should be of the same " +
+                "dimensionality. The Euclidean distance is taken as the distance metric. " +
+                "The algorithm resembles Sequential K-Means Clustering at " +
+                "https://www.cs.princeton.edu/courses/archive/fall08/cos436/Duda/C/sk_means.htm ",
+        parameters = {
+                @Parameter(
+                        name = "model.name",
+                        description = "The name for the model that is going to be created/reused for prediction",
+                        type = {DataType.STRING}
+                ),
+                @Parameter(
+                        name = "decay.rate",
+                        description = "this is the decay rate of old data compared to new data. " +
+                                "Value of this will be in [0,1]. 0 means only old data used and" +
+                                "1 will mean that only new data is used",
+                        optional = true,
+                        type = {DataType.FLOAT},
+                        defaultValue = "0.01f"
+                ),
+                @Parameter(
+                        name = "number.of.clusters",
+                        description = "The assumed number of natural clusters (numberOfClusters) in the data set.",
+                        type = {DataType.INT}
+                ),
+                @Parameter(
+                        name = "coordinate.values",
+                        description = "This is a variable length argument. Depending on the dimensionality of data " +
+                                "points we will receive coordinates along each axis.",
+                        type = {DataType.DOUBLE} //how to give multiple types?
+                )
+
+        },
+        returnAttributes = {
+                @ReturnAttribute(
+                        name = "euclideanDistanceToClosestCentroid",
+                        description = "Represents the Euclidean distance between the current data point and the " +
+                                "closest centroid.",
+                        type = {DataType.DOUBLE}
+                ),
+                @ReturnAttribute(
+                        name = "closestCentroidCoordinatei",
+                        description = "This is a variable length attribute. Depending on the dimensionality(d) " +
+                                "we will return closestCentroidCoordinate1 to closestCentroidCoordinated",
+                        type = {DataType.DOUBLE}
+                )
+        },
+        examples = {
+                @Example(
+                        syntax = "from InputStream#streamingML:KMeansIncremental(modelName, numberOfClusters, " +
```

Comment:
Better to define the stream; check https://github.com/wso2-extensions/siddhi-execution-streamingml/blob/master/component/src/main/java/org/wso2/extension/siddhi/execution/streamingml/classification/perceptron/PerceptronClassifierUpdaterStreamProcessorExtension.java#L75

Reply:
done